rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMController.java

```diff
@@ -684,7 +666,120 @@ public void onFailure(Throwable caught) {
         model.setNumPhysicalHosts(numPhysical);
         model.setNumVirtualHosts(numVirtual);
 
-        return ns;
+    }
+
+    private void recalculateStatistics() {
+        model.setNumBusy(0);
+        model.setNumConfiguring(0);
+        model.setNumDeploying(0);
+        model.setNumDown(0);
+        model.setNumFree(0);
+        model.setNumLost(0);
+        model.setNumToBeRemoved(0);
+        model.setNumLocked(0);
+        model.setNumDeployedNodeSources(0);
+        model.setNumUndeployedNodeSources(0);
+
+        for (NodeSource nodeSource : model.getNodeSources().values()) {
+            switch (nodeSource.getNodeSourceStatus()) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a default case to this switch. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)